wcfsetup/install/files/lib/system/request/RequestHandler.class.php

```diff
@@ -106,6 +105,88 @@ public function handle($application = 'wcf', $isACPRequest = false)
         }
     }
 
+    /**
+     * Splits the given array of cache-control values at commas, while properly
+     * taking into account that each value might itself contain commas within a
+     * quoted string.
+     */
+    private function splitCacheControl(array $values): \Iterator
+    {
+        foreach ($values as $value) {
+            $isQuoted = false;
+            $result = '';
+
+            for ($i = 0, $len = \strlen($value); $i < $len; $i++) {
+                $char = $value[$i];
+                if (!$isQuoted && $char === ',') {
+                    yield \trim($result);
+
+                    $isQuoted = false;
+                    $result = '';
+
+                    continue;
+                }
+
+                if ($isQuoted && $char === '\\') {
+                    $result .= $char;
+                    $i++;
+
+                    if ($i < $len) {
+                        $result .= $value[$i];
+
+                        continue;
+                    }
+                }
+
+                if ($char === '"') {
+                    $isQuoted = !$isQuoted;
+                }
+
+                $result .= $char;
+            }
+
+            if ($result !== '') {
+                yield \trim($result);
+            }
+        }
+    }
+
+    /**
+     * @since 5.5
+     */
+    private function sendPsr7Response(ResponseInterface $response)
+    {
+        // Storing responses in a shared cache is unsafe, because they all contain session specific information.
+        // Add the 'private' value to the cache-control header and remove any 'public' value.
+        $cacheControl = [];
```

Comment:
```suggestion
        $cacheControl = ['private'];
```

The `'private'` value is always set, moving it into the initialization makes more sense. It also provides a better explanation for line 165 which talks about the "duplication" of this value.

Reply:
Ah, indeed. The code initially used `array_filter`, until I added `splitCacheControl` (which is just a generator / iterator).